### DISCUSSION TITLE
Retain task order when quitting and re-opening Taskbar (e.g. after a restart)

### DISCUSSION THREAD

**Discussion by gavingt** [lawand-dot-io/taskbar#64] [Feature requests]
After restarting my Mac, the order of the tasks in the taskbar is different than how it was before the restart.

**Comment by lawand-dot-io** [comment]
Hello. Thanks for this post. This is actually not a bug as Taskbar doesn't save the list of windows, and rather reads them again on the next restart. This is by design because windows might open or close between quitting Taskbar and running it again. Also, in a typical use case restarts would be infrequent. In any case it's not up to me to tell users how to interact with macOS and Taskbar, so I will convert this to a feature request, and as always feature requests that receive more upvotes are more likely to be implemented as I have to prioritize the requests.

**Reply by gavingt** [reply]
Fair enough! I agree that it's not a huge deal. I just created the ticket in response to a [Reddit post](https://www.reddit.com/r/MacOS/comments/1e8h0en/comment/lea871f/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button).

**Reply by GsHeri** [reply]
just save/load the list, and ALSO read the windows, add the ones not present in the list to the end. problem solved :)
